gobblin-modules/gobblin-couchbase/src/main/java/gobblin/couchbase/writer/CouchbaseWriter.java

```diff
@@ -103,38 +128,132 @@ public CouchbaseWriter(CouchbaseEnvironment couchbaseEnvironment, Config config)
     _bucket = _cluster.openBucket(bucketName, password,
         Collections.<Transcoder<? extends Document, ?>>singletonList(_tupleDocumentTranscoder));
 
+
+    _operationTimeout = 2000;
+    _operationTimeunit = TimeUnit.MILLISECONDS;
+
   }
 
 
+  @VisibleForTesting
   Bucket getBucket()
   {
     return _bucket;
   }
 
+
   @Override
-  public void cleanup()
-      throws IOException {
+  public Future<WriteResponse> write(final D record, final WriteCallback callback) {
+    if (record instanceof TupleDocument) {
```

Comment:
JsonDocument works as well.  Point taken about understanding the behavior of this when non-compliant types are sent down.